### ISSUE TITLE
Don't use "blacklist" or "whitelist" in our codebase

### ISSUE THREAD

**Issue by nicksnyder** [sourcegraph/sourcegraph#11389]
We are against racism. "blacklist" and "whitelist" carry the implication that black is bad and white is good.

We should use words that directly describe what we mean.
backlist -> blocklist
whitelist -> allowlist

Other teams are changing this too: https://9to5google.com/2020/06/07/google-chrome-blacklist-blocklist-more-inclusive/

**Comment by blacklistisnotracist** [comment]
@nicksnyder 

I won't argue with you about which of whitelist vs allowlist is more descriptive. It's a subjective preference and I do agree with employing allowlist to some extent because it conveys meaning clearer. But never because it carries racist connotations.

Calling out whitelist/blacklist being racist words is simply wrong because these words have nothing to do with racism.

The color black has association with darkness. This is inevitable as humans are diurnal creatures and are scared of darkness because we can't see things well and predators might come out of the dark.

It would make more sense to stop calling groups of people with colors.
https://github.com/blacklistisnotracist/blacklist-is-not-a-racist-word/blob/master/README.md

In summary, my point is, they don't carry racist connotations so stop spreading the lie. It's not only wrong but also reinforce the wrong meaning to the public. This is simply ridiculous.

**Comment by felixfbecker** [comment]
I'd like to cite Go's commit message:

> all: replace usages of whitelist/blacklist and master/slave
> 
> There's been plenty of discussion on the usage of these terms in tech.
> I'm not trying to have yet another debate. It's clear that there are
> people who are hurt by them and who are made to feel unwelcome by their
> use due not to technical reasons but to their historical and social
> context. That's simply enough reason to replace them.
> 
> Anyway, allowlist and blocklist are more self-explanatory than whitelist
> and blacklist, so this change has negative cost.

https://go-review.googlesource.com/c/go/+/236857/

**Comment by blacklistisnotracist** [comment]
@felixfbecker 

> It's clear that there are people who are hurt by them

This is not clear at all to me. All of those who make this change is actually white people who claim there are racist connotations without presenting any evidence.

**Comment by blacklistisnotracist** [comment]
A good watch for you: https://www.youtube.com/watch?v=GNXm7juuM-8

**Comment by keegancsmith** [comment]
Was just reading the git mailing list as one does, and they are considering changing the default branch name. Linked to this tweet https://twitter.com/mislav/status/1270388510684598272

I quite like trunk since that is what SVN uses and we are effectively like SVN in that we have a central repository. Alternatively mercurial's default branch is "default", so seems like a decent name. I see from discussions on the mailing list "main" is quite popular. We could also use "develop" since we develop against the default branch.

**Comment by felixfbecker** [comment]
`release` would also be a possibility, given it's the branch we automatically release from on every merge. Otherwise I also like `main`. `trunk` gives me negative connotations, like stuffing a car trunk full with way too much stuff before going on a holiday trip.

**Comment by nicksnyder** [comment]
Agree about renaming our default branch.  I like`main` and `default` the best. `release` would cause confusion with release branches that already exist. `trunk` is too SVN and opaque for my taste.

**Comment by keegancsmith** [comment]
> Agree about renaming our default branch.  I like`main` and `default` the best. `release` would cause confusion with release branches that already exist. `trunk` is too SVN and opaque for my taste.

I vote "main". Conveys the meaning best. Note: our development
philosophy is Trunk Based Development [1]. I'm also nostalgic about
calling it trunk. But main makes the most sense.

[1]: https://trunkbaseddevelopment.com/

**Comment by keegancsmith** [comment]
Was pretty quick to update the very few places we hardcoded default. Sent this PR which updates our dev tooling to treat main as the default branch. Not a guarantee that there won't be some other fall out by making this change, but relatively safe :) https://github.com/sourcegraph/sourcegraph/pull/11453

**Comment by nicksnyder** [comment]
I created a separate issue for the branch rename work: https://github.com/sourcegraph/sourcegraph/issues/11464

**Comment by uwedeportivo** [comment]
Dear all,

This is your release captain speaking. 🚂🚂🚂

Branch cut for the **3.17 release is scheduled for tomorrow**.

Is this issue / PR going to make it in time? Please change the milestone accordingly.
When in doubt, reach out!

Thank you

**Comment by daxmc99** [comment]
Dear all,

This is your release captain speaking. 🚂🚂🚂

Branch cut for the **3.18 release is scheduled for tomorrow**.

Is this issue / PR going to make it in time? Please change the milestone accordingly.
When in doubt, reach out!

Thank you

**Comment by nicksnyder** [comment]
I have cleaned up as much of our codebase as I can (apparently GitHub does not show all the changes I made in the timeline of this issue because this issue is locked).

https://sourcegraph.sgdev.org/search?q=blacklist%7Cwhitelist+r:%5Esourcegraph/+count:1000+-file:%28%5E%7C/%29vendor/+-r:sourcegraph/ghdump&patternType=regexp

The biggest remaining work item is https://github.com/sourcegraph/sourcegraph/issues/11394, which I have opened a PR to address.

Closing this.
